permission-portal-frontend/client/src/screens/CodeValidations.js

```diff
@@ -8,70 +7,62 @@ import { withStore } from '../store'
       <h1>Positive Test Validations</h1>
       <div id="actions-box" className="gray-background">
-        <div className="action-section">
+        <div className="validation-container">
           <div className="section-heading-container">
-            <h2 className="section-heading">Validation Code</h2>
+            <h2>Validation Code</h2>
             <div className="tooltip">
               <img src={info_icon} alt="info" />
               <div className="tooltip-msg">
-                {/* to replace tooltip text here: */}
-                <div className="tooltip-title">This the tooltip title</div>
+                {/* TODO replace tooltip text */}
+                <div className="tooltip-title">Test Validation Codes</div>
                 <div className="tooltip-body">
                   Quisque sagittis, vel hendrerit consectetur tincidunt senectus. Feugiat aenean nunc, tempus tempus,
                   porta nibh. Nunc id donec enim ut potenti risus amet amet.
                 </div>
               </div>
             </div>
           </div>
-          <p className="section-description">
-            Enter the positive test validation code the user gave to you over the phone.
-          </p>
-          <input type="text" placeholder="281-177-9" onChange={(e) => setCode(e.target.value)}></input>
-        </div>
-
-        <div className="action-section">
-          <h2 className="section-heading">Tracing Start Date</h2>
-          <p className="section-description">
-            Enter facilisis etiam. Felis sed blandit in lacus urna et, arcu notiar, dui, lorem.
-          </p>
-          <input type="date" onChange={(e) => setDate(e.target.value)}></input>
-        </div>
-
-        <div id="white-box" className="white-background">
-          {/* circle loading graphic demo working:
-            To do: trigger this upon click and loading of data.  Will this data processing take sufficiently long to justify having this loading graphic at all?
-          */}
-          {/* <div id="progress-container">
-            <CircularProgress />
-          </div> */}
-
-          <Button
-            id={`verify-code-btn${date === '' || code === '' ? '-disabled' : ''}`}
-            onClick={() => confirmedToast.current.show()}
-            disabled={date === '' || code === ''}
-          >
-            Verify Code
-          </Button>
+          <div className="validation-text">
+            Provide this validation code to the person you want to verify over the phone
+          </div>
+          <div className="code-box">{code}</div>
+          <PendingOperationButton className="save-button generate-button" operation={genNewCode}>
+            Generate New Code
+          </PendingOperationButton>
         </div>
       </div>
-      <Toast ref={confirmedToast} isSuccess={true} message="Code verification confirmed" />
+      <Toast ref={confirmedToast} isSuccess={toastInfo.success} message={toastInfo.msg} />
```

Comment:
I think ref is an antipattern in react and discouraged. In other pages we've been using state.

Reply:
After [PR 193](https://github.com/covid19risk/permission-portal/pull/193) this is the way all the toasts work across the app, so I was just following those.

Reply:
I changed all the toasts to refs. Happy to discuss alternatives.

Reply:
I'd say I'm fine if it's consistent and works without issue.